src/main/java/org/hypertrace/gradle/code/style/CodeStylePlugin.java

```diff
@@ -0,0 +1,50 @@
+package org.hypertrace.gradle.code.style;
+
+import com.diffplug.gradle.spotless.SpotlessExtension;
+import com.diffplug.gradle.spotless.SpotlessPlugin;
+import java.util.Map;
+import javax.annotation.Nonnull;
+import org.gradle.api.Plugin;
+import org.gradle.api.Project;
+import org.gradle.api.plugins.PluginContainer;
+
+public class CodeStylePlugin implements Plugin<Project> {
+
+  @Override
+  public void apply(@Nonnull Project target) {
+    configureCodeStyle(target);
+  }
+
+  private void configureCodeStyle(Project project) {
+    PluginContainer pluginContainer = project.getPlugins();
+    pluginContainer.apply(SpotlessPlugin.class);
+
+    SpotlessExtension spotlessExtension =
+        project.getExtensions().getByType(SpotlessExtension.class);
+    configureFormatting(spotlessExtension);
```

Comment:
One thing we generally do when using this is missing I believe - 
```kts
  tasks {
    check {
      dependsOn("spotlessCheck")
    }
  }
```

That'll come out to something like 
```java
    project
        .getTasks()
        .named(LifecycleBasePlugin.CHECK_TASK_NAME)
        .configure(task -> task.dependsOn(compatibilityTask));
```

Reply:
spotless extension takes care of it automatically https://github.com/diffplug/spotless/blob/a30553386acfa24f0d97f587249c38f44feb79b6/plugin-gradle/src/main/java/com/diffplug/gradle/spotless/SpotlessExtensionImpl.java#L49